lms/djangoapps/courseware/views/views.py

```diff
@@ -393,9 +396,13 @@ def course_info(request, course_id):
 
         response = render_to_response('courseware/info.html', context)
         if store_upgrade_cookie:
+            cookie_value = course_id
+            if upgrade_cookie_name in request.COOKIES and course_id not in request.COOKIES[upgrade_cookie_name]:
```

Comment:
I think there is a bug in this logic. If course_id IS in request.COOKIES[upgrade_cookie_name], then the existing cookie value will be wiped out with just course_id.